# VoC and Customer Analysis

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

**Google Play Review Dataset**

In [10]:
# read google play review data -> save dataframe to rev_df

rev_df = pd.read_csv('../data/raw/Google Play Reviews/reviews.csv')

Let's get a sense of the structure of our data..

In [11]:
rev_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12495 entries, 0 to 12494
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   reviewId              12495 non-null  object
 1   userName              12495 non-null  object
 2   userImage             12495 non-null  object
 3   content               12495 non-null  object
 4   score                 12495 non-null  int64 
 5   thumbsUpCount         12495 non-null  int64 
 6   reviewCreatedVersion  10333 non-null  object
 7   at                    12495 non-null  object
 8   replyContent          5818 non-null   object
 9   repliedAt             5818 non-null   object
 10  sortOrder             12495 non-null  object
 11  appId                 12495 non-null  object
dtypes: int64(2), object(10)
memory usage: 1.1+ MB


In [14]:
# 1. check the shape of the data
rev_df.shape

(12495, 12)

There are 12 total columns, and 12,495 rows of data.

In [15]:
# 2. check the datatypes
rev_df.head(2)

,reviewId,userName,userImage,content,score,thumbsUpCount,reviewCreatedVersion,at,replyContent,repliedAt,sortOrder,appId
0,gp:AOqpTOEhZuqSqqWnaKRgv-9ABYdajFUB0WugPGh-SG-...,Eric Tie,https://play-lh.googleusercontent.com/a-/AOh14...,I cannot open the app anymore,1,0,5.4.0.6,2020-10-27 21:24:41,NaN,NaN,newest,com.anydo
1,gp:AOqpTOH0WP4IQKBZ2LrdNmFy_YmpPCVrV3diEU9KGm3...,john alpha,https://play-lh.googleusercontent.com/a-/AOh14...,I have been begging for a refund from this app...,1,0,NaN,2020-10-27 14:03:28,"Please note that from checking our records, yo...",2020-10-27 15:05:52,newest,com.anydo


Some columns are do not have their right datatypes. Some of these datatypes are not memory optimal, and currently consuming 1MB+ of memory storage. 

Let's fix the datatype of the columns

In [20]:
# explore 'score' column
print('These are the unique values in "score" column', rev_df['score'].unique())

print('The datatype of this column is ', rev_df['score'].dtype)

These are the unique values in "score" column [1 2 3 4 5]
The datatype of this column is  int64


In [21]:
# explore 'thumbsUpCount' column
print('These are the unique values in "score" column', rev_df['thumbsUpCount'].unique())

print('The datatype of this column is ', rev_df['thumbsUpCount'].dtype)

These are the unique values in "score" column [  0   1  17  20  59   2   3  86  10  61  18  22  19  29 127  13  30  21
  81  16  11   5  60   7  24  12   8 107   6   4  14  49  45  74   9  35
  71  51  55  31  96  33  70 156  23 179 175  25 101 231 148 116  39  37
  75  83  65 115  38  27  80  92  40 270 184 157  73 173 282  26  36  15
  32  52 319  53 118  28  79 119  62 138  46  63  34  72  91  47  88  50
 142  44  41  43  54  48  90  84 178  77 290 121  89 100  85  98  87 281
 187 133 103 278 177  93 117  66 125  64 113  97  69  58 109  42 160  94
 205 105  99  57  78 136 143 124 241 162  68 397 339]
The datatype of this column is  int64


In [25]:
# explore 'at' column
print('The datatype of this column is', rev_df['at'].dtype)

The datatype of this column is object


This is a datetime dtype. Let's convert it to its right type

In [26]:
rev_df['at'] = pd.to_datetime(rev_df['at'], errors='coerce')

In [30]:
# explore 'repliedAt' column
print('These are the unique values in "score" column', rev_df['repliedAt'].dtype)

These are the unique values in "score" column object


In [ ]:
# change repliedAt to datetime
rev_df['at'] = pd.to_datetime(rev_df['at'], errors='coerce')